mixins/StoreMixin.js

```diff
@@ -18,53 +18,53 @@ var ListenersDescription = {
 };
 
 var StoreMixin = {
-  componentDidMount: function () {
-    if (this.store_listeners) {
-      // Create a map of listeners, becomes useful later
-      var storesListeners = {};
-
-      // Merges options for each store listener with
-      // the ListenersDescription definition above
-      this.store_listeners.forEach(function (listener) {
-        if (typeof listener === 'string') {
-          if (!ListenersDescription[listener]) {
-            return;
-          }
-
-          // Use all defaults
-          storesListeners[listener] = Util.clone(ListenersDescription[listener]);
-        } else {
-          var storeName = listener.name;
-          var events = listener.events;
-
-          if (!ListenersDescription[storeName]) {
-            return;
-          }
-
-          // Populate events by key. For example, a component
-          // may only want to listen for 'success' events
-          if (events) {
-            listener.events = {};
-            events.forEach(function (event) {
-              listener.events[event] =
-                ListenersDescription[storeName].events[event];
-            });
-          }
-
-          storesListeners[storeName] = Util.extend(
-            {}, ListenersDescription[storeName], listener
-          );
+  store_setListeners: function (storeListeners) {
```

Comment:
what do you think about calling this `store_initializeListeners`? `setListeners` to me sounds a lot of like `addListeners` whereas if we had the word initialize in there, it seems like something that should be called once per use